src/Contrib/Otlp/ProtobufSerializer.php

```diff
@@ -50,10 +50,9 @@ public function serialize(Message $message): string
             case self::PROTOBUF:
                 return $message->serializeToString();
             case self::JSON:
-                //@todo https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/otlp.md#json-protobuf-encoding
-                return $message->serializeToJsonString();
+                return self::fixJsonOutput($message->serializeToJsonString());
```

Comment:
That's 1 option I mentioned in #969. Do you think that it's good enough or at some point we would like to rewrite whole serializer?

Reply:
I think we probably will need to write a custom serializer at some point, given the other documented differences between json and other protobuf. I might go ask around in the collector SIG for advice here.
In the meantime, I think I'll refactor this PR to move most of this logic into a `JsonSerializer` class.